packages/components-react/heading-react/src/heading.test.tsx

```diff
@@ -1,42 +1,214 @@
+
+  it('renders a HTML h4 element', () => {
     const { container } = render(<Heading level={4} />);
-    const heading = container.querySelector('h4');
+    const heading = container.querySelector('h4:only-child');
 
     expect(heading).toBeInTheDocument();
   });
-  it('renders a heading level 5', () => {
+
+  it('renders a HTML h5 element', () => {
     const { container } = render(<Heading level={5} />);
-    const heading = container.querySelector('h5');
+    const heading = container.querySelector('h5:only-child');
 
     expect(heading).toBeInTheDocument();
   });
-  it('renders a heading level 6', () => {
+
+  it('renders a HTML h6 element', () => {
     const { container } = render(<Heading level={6} />);
-    const heading = container.querySelector('h6');
+    const heading = container.querySelector('h6:only-child');
 
     expect(heading).toBeInTheDocument();
   });
+  it('renders a HTML h1 element with the correct className', () => {
+    const { container } = render(<Heading level={1} />);
+    const heading = container.querySelector('h1:only-child');
+
+    expect(heading).toHaveClass('nl-heading-1');
+  });
+
+  it('renders a HTML h2 element with the correct className', () => {
+    const { container } = render(<Heading level={2} />);
+    const heading = container.querySelector('h2:only-child');
+
+    expect(heading).toHaveClass('nl-heading-2');
+  });
+
+  it('renders a HTML h3 element with the correct className', () => {
+    const { container } = render(<Heading level={3} />);
+    const heading = container.querySelector('h3:only-child');
+
+    expect(heading).toHaveClass('nl-heading-3');
+  });
+
+  it('renders a HTML h4 element with the correct className', () => {
+    const { container } = render(<Heading level={4} />);
+    const heading = container.querySelector('h4:only-child');
+
+    expect(heading).toHaveClass('nl-heading-4');
+  });
+
+  it('renders a HTML h5 element with the correct className', () => {
+    const { container } = render(<Heading level={5} />);
+    const heading = container.querySelector('h5:only-child');
+
+    expect(heading).toHaveClass('nl-heading-5');
+  });
+
+  it('renders a HTML h6 element with the correct className', () => {
+    const { container } = render(<Heading level={6} />);
+    const heading = container.querySelector('h6:only-child');
+
+    expect(heading).toHaveClass('nl-heading-6');
+  });
+
+  it('renders rich text content', () => {
+    render(
+      <Heading level={1}>
+        <strong>strong</strong>
+      </Heading>,
+    );
+    const heading = screen.getByRole('heading', { level: 1 });
+    const strong = heading.querySelector('strong');
+
+    expect(strong).toBeInTheDocument();
+  });
+
+  it('renders an inaccessible hidden heading', () => {
+    render(<Heading level={1} hidden />);
+    const heading = screen.getByRole('heading', { hidden: true });
+
+    expect(heading).not.toBeVisible();
+  });
+
+  it('forwards React refs to the heading', () => {
```

Comment:
misschien moeten we hier expliciet maken dat het naar de HTML heading element is, vanwege de `<HTMLHeadingElement>`